### ISSUE TITLE
Controller Integration Tests

### ISSUE THREAD

**Issue by seanjparker** [hacksoc-manchester/Hackathon_Attendee_Hub#114]
We have a very low test coverage when it comes to the controllers, currently around 27% overall. We need to write more tests for both the user and hardware controllers. The other controllers need new tests to be written.

**Comment by kzalys** [comment]
Why integration tests? We should be able to mock out all of the controllers' dependencies with the new architecture and make unit tests.

**Comment by seanjparker** [comment]
Currently, we have some controller tests, we send HTTP requests to a version of the built application which doesn't just tests the controllers but also hits the services, routers and database. I suppose these would be more like system tests since we are testing every component together. We can also do unit tests for the controllers which is a good idea.

**Comment by kzalys** [comment]
Yeah I think if we're doing new tests for the controllers, they should be unit tests (they're probably quicker to write than e2e tests at this point). Having a few end-to-end tests for the entire system might be a good idea as well but if the controllers are covered by unit tests, I'm not too sure it would make sense investing a lot of effort into end-to-end tests since we'd just be checking the same test cases as in the unit tests.
